.github/workflows/build_and_deploy.yml

```diff
@@ -509,23 +551,25 @@ jobs:
           path: /home/runner/.npm/_logs/*
 
   deployExamples:
+    if: ${{ needs.deploy-target.outputs.value != 'automated-preview' }}
     name: Deploy examples
     runs-on: ubuntu-latest
-    needs: [build]
+    needs: [build, deploy-target]
     steps:
+      - run: echo '${{ needs.deploy-target.outputs.value }}'
       - uses: actions/checkout@v4
         with:
           fetch-depth: 25
       - name: Install Vercel CLI
         run: npm i -g vercel@latest
       - name: Deploy preview examples
-        if: ${{ needs.build.outputs.isRelease != 'true' }}
+        if: ${{ needs.deploy-target.outputs.value != 'production' }}
```

Comment:
should this whole job be gated on a deploy-target of `production` rather than the individual steps?